zzzz_modular_occulus/code/game/objects/items/devices/cloakingfield.dm

```diff
@@ -0,0 +1,187 @@
+				src.anti_spam = world.time
+				my_gen.turn_off()
+			else
+				if (src.anti_spam && world.time < src.anti_spam + 100)
+					to_chat(usr,"The cloaking field generator is recharging!", "red")
+					return
+				src.anti_spam = world.time
+				my_gen.turn_on()
+		else
+			to_chat(usr, SPAN_NOTICE("<span class='alert'>No signal detected. Swipe remote on a cloaking generator to establish a connection.</span>"))
+	return
+
+/obj/item/remote/cloak_gen/verb/set_pattern()
+	set src in view(1)
+	if (!isliving(usr) || !my_gen) return
+	var/input = input(usr,"Select cloaking pattern:","Set pattern","Noise") in list("Noise","Linear","Chaos","Cubic","Interference","Rotating")
+	switch(input)
+		if("Linear")
+			my_gen.icon_to_use = "noise1"
+		if("Noise")
+			my_gen.icon_to_use = "noise2"
+		if("Chaos")
+			my_gen.icon_to_use = "noise3"
+		if("Cubic")
+			my_gen.icon_to_use = "noise4"
+		if("Interference")
+			my_gen.icon_to_use = "noise5"
+		if("Rotating")
+			my_gen.icon_to_use = "noise6"
+	if(my_gen.active)
+		my_gen.turn_off()
+		my_gen.turn_on()
+	to_chat(usr, SPAN_NOTICE("<span class='notice'>You set the pattern to '[input]'.</span>"))
+
+/obj/item/remote/cloak_gen/verb/set_range()
+	set src in view(1)
+	if (!isliving(usr) || !my_gen) return
+	var/input = input(usr,"Range 0-[my_gen.maxrange]:","Set range",my_gen.range) as num
+	if(input > my_gen.maxrange || input < 0)
+		to_chat(usr, SPAN_NOTICE("<span class='alert'>Invalid setting.</span>"))
+		return
+	my_gen.range = input
+	if(my_gen.active)
+		my_gen.turn_off()
+		my_gen.turn_on()
+	to_chat(usr, SPAN_NOTICE("<span class='notice'>You set the range to [my_gen.range].</span>"))
+
+/obj/item/remote/cloak_gen/verb/increase_range()
+	set src in view(1)
+	if (!isliving(usr) || !my_gen) return
+	if (my_gen.range + 1 > my_gen.maxrange)
+		to_chat(usr, SPAN_NOTICE("<span class='alert'>Maximum range reached ([my_gen.maxrange]).</span>"))
+		return
+	my_gen.range++
+	if(my_gen.active)
+		my_gen.turn_off()
+		my_gen.turn_on()
+	to_chat(usr, SPAN_NOTICE("<span class='notice'>You set the range to [my_gen.range].</span>"))
+
+/obj/item/remote/cloak_gen/verb/decrease_range()
+	set src in view(1)
+	if (!isliving(usr) || !my_gen) return
+	if (my_gen.range - 1 < 0)
+		to_chat(usr, SPAN_NOTICE("<span class='alert'>Minimum range reached (0).</span>"))
+		return
+	my_gen.range--
+	if(my_gen.active)
+		my_gen.turn_off()
+		my_gen.turn_on()
+	to_chat(usr, SPAN_NOTICE("<span class='notice'>You set the range to [my_gen.range].</span>"))
+
+/obj/item/remote/cloak_gen/verb/turn_on()
+	set src in view(1)
+	if (!isliving(usr) || !my_gen || my_gen.active) return
+	my_gen.turn_on()
+	to_chat(usr, SPAN_NOTICE("<span class='notice'>You turn the cloaking field generator on.</span>"))
+
+/obj/item/remote/cloak_gen/verb/turn_off()
+	set src in view(1)
+	if (!isliving(usr) || !my_gen || !my_gen.active) return
+	my_gen.turn_off()
+	to_chat(usr, SPAN_NOTICE("<span class='notice'>You turn the cloaking field generator off.</span>"))
```

Comment:
This was not resolved.

Reply:
This doesn't need to be combined.

Reply:
It would be preferable to make a single verb to toggle over two separate verbs. To keep the already cluttered verb panel from getting even more over-crowded .

Reply:
Guh. Alright.